inst/data_survival.R

```diff
@@ -0,0 +1,56 @@
+# Load required libraries
+library(TreeSummarizedExperiment)
+library(S4Vectors)
+library(mia)
+
+data("data_survival", package = "coda4microbiome")
+data_survival <- x
+assay_matrix <- as.matrix(data_survival)
+assay_matrix <- t(assay_matrix)
+taxa_names <- rownames(assay_matrix)
+
+# Function to Parse Taxonomy Labels
+parse_taxonomy <- function(taxa_names) {
+  taxonomy_df <- data.frame(TaxonID = taxa_names, stringsAsFactors = FALSE)
+  
+  # Add structured taxonomy placeholders
+  taxonomy_df$Kingdom <- NA  
+  taxonomy_df$Phylum <- NA
+  taxonomy_df$Class <- NA
+  taxonomy_df$Order <- NA
+  taxonomy_df$Family <- NA
+  taxonomy_df$Genus <- NA
+  
+  # Fill known taxonomic ranks
+  for (i in seq_along(taxa_names)) {
+    taxon <- taxa_names[i]
+    if (grepl("^g__", taxon)) {
+      taxonomy_df$Genus[i] <- sub("^g__", "", taxon)
+    } else if (grepl("^f__", taxon)) {
+      taxonomy_df$Family[i] <- sub("^f__", "", taxon)
+    } else if (grepl("^o__", taxon)) {
+      taxonomy_df$Order[i] <- sub("^o__", "", taxon)
+    }
+  }
```

Comment:
As this is a script generating data and this is not regularly used, looping is ok. However, just for practice you could vectorize this.

I think this might work like this:

```
ind <- grepl("^g__", taxa_names)
taxonomy_df[ind, "Genus"] <- gsub("^g__", "", taxa_names[ind])
```

However, this assumes that the dataset only includes Genus, Family or Order level taxa? Is that the case?